src/platform/plugins/shared/discover/public/embeddable/get_search_embeddable_factory.tsx

```diff
@@ -57,61 +54,58 @@ export const getSearchEmbeddableFactory = ({
+        runtimeState.nonPersistedDisplayOptions?.solutionNavIdOverride ??
         (await firstValueFrom(discoverServices.core.chrome.getActiveSolutionNavId$()));
       const { getRenderAppWrapper } = await discoverServices.profilesManager.resolveRootProfile({
         solutionNavId,
       });
       const AppWrapper = getRenderAppWrapper?.(BaseAppWrapper) ?? BaseAppWrapper;
 
       /** Specific by-reference state */
-      const savedObjectId$ = new BehaviorSubject<string | undefined>(initialState?.savedObjectId);
-      const defaultTitle$ = new BehaviorSubject<string | undefined>(initialState?.savedObjectTitle);
+      const savedObjectId$ = new BehaviorSubject<string | undefined>(runtimeState?.savedObjectId);
+      const defaultTitle$ = new BehaviorSubject<string | undefined>(runtimeState?.savedObjectTitle);
       const defaultDescription$ = new BehaviorSubject<string | undefined>(
-        initialState?.savedObjectDescription
+        runtimeState?.savedObjectDescription
       );
 
-      /** By-value SavedSearchComponent package (non-dashboard contexts) state, to adhere to the comparator contract of an embeddable. */
-      const nonPersistedDisplayOptions$ = new BehaviorSubject<
-        NonPersistedDisplayOptions | undefined
-      >(initialState?.nonPersistedDisplayOptions);
-
       /** All other state */
       const blockingError$ = new BehaviorSubject<Error | undefined>(undefined);
       const dataLoading$ = new BehaviorSubject<boolean | undefined>(true);
       const fetchContext$ = new BehaviorSubject<FetchContext | undefined>(undefined);
       const fetchWarnings$ = new BehaviorSubject<SearchResponseIncompleteWarning[]>([]);
 
       /** Build API */
-      const titleManager = initializeTitleManager(initialState);
-      const timeRange = initializeTimeRange(initialState);
-      const dynamicActionsApi =
-        discoverServices.embeddableEnhanced?.initializeReactEmbeddableDynamicActions(
+      const titleManager = initializeTitleManager(initialState.rawState);
+      const timeRangeManager = initializeTimeRangeManager(initialState.rawState);
+      const dynamicActionsManager =
+        discoverServices.embeddableEnhanced?.initializeEmbeddableDynamicActions(
           uuid,
           () => titleManager.api.title$.getValue(),
-          initialState
+          initialState.rawState
         );
-      const maybeStopDynamicActions = dynamicActionsApi?.startDynamicActions();
-      const searchEmbeddable = await initializeSearchEmbeddableApi(initialState, {
+      const maybeStopDynamicActions = dynamicActionsManager?.startDynamicActions();
+      const searchEmbeddable = await initializeSearchEmbeddableApi(runtimeState, {
         discoverServices,
       });
       const unsubscribeFromFetch = initializeFetch({
         api: {
           parentApi,
           ...titleManager.api,
-          ...timeRange.api,
+          ...timeRangeManager.api,
+          defaultTitle$,
```

Comment:
Why the addition of `defaultTitle$` here?

Reply:
If you look at the implementation of [initializeFetch](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/discover/public/embeddable/initialize_fetch.ts), it calls `api.defaultTitle$?.getValue()`, so `defaultTitle$` should be passed into `initializeFetch`.